com.hazelcast.jet.core.metrics.JobMetrics_MiscTest.when_jobFailedBeforeStarted_then_emptyMetrics[1: mode=client]

_master_ (commit bb53df18aa0fd10f2e347a83acd7949a26a4dd52)

Failed on Oracle JDK8: http://jenkins.hazelcast.com/job/jet-oss-master-oracle-jdk8/97/testReport/junit/com.hazelcast.jet.core.metrics/JobMetrics_MiscTest/when_jobFailedBeforeStarted_then_emptyMetrics_1__mode_client_/

Stacktrace:
```
org.junit.ComparisonFailure: 
expected exception not found in causes chain expected:<java.[lang.RuntimeException: foo]> but was:<java.[util.concurrent.CompletionException: com.hazelcast.jet.JetException: java.lang.RuntimeException: foo
	at java.util.concurrent.CompletableFuture.reportJoin(CompletableFuture.java:375)
	at java.util.concurrent.CompletableFuture.join(CompletableFuture.java:1934)
	at com.hazelcast.jet.Job.join(Job.java:137)
	at com.hazelcast.jet.core.metrics.JobMetrics_MiscTest.when_jobFailedBeforeStarted_then_emptyMetrics(JobMetrics_MiscTest.java:126)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.hazelcast.jet.JetException: java.lang.RuntimeException: foo
	at com.hazelcast.jet.impl.JobResult.getFailureAsThrowable(JobResult.java:113)
	at com.hazelcast.jet.impl.JobResult.asCompletableFuture(JobResult.java:125)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$callWithJob$31(JobCoordinationService.java:553)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$42(JobCoordinationService.java:963)
	at com.hazelcast.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:67)
	at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:227)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
	at ------ submitted from ------.(Unknown Source)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:126)
	at com.hazelcast.spi.impl.AbstractInvocationFuture$1.run(AbstractInvocationFuture.java:251)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at com.hazelcast.jet.core.TestUtil.assertExceptionInCauses(TestUtil.java:68)
	at com.hazelcast.jet.core.metrics.JobMetrics_MiscTest.when_jobFailedBeforeStarted_then_emptyMetrics(JobMetrics_MiscTest.java:129)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
Hiccups measured while running test 'when_jobFailedBeforeStarted_then_emptyMetrics[1: mode=client](com.hazelcast.jet.core.metrics.JobMetrics_MiscTest):'
13:27:25, accumulated pauses: 125 ms, max pause: 108 ms, pauses over 1000 ms: 0
```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/JobMetrics_MiscTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/JobMetrics_MiscTest.java
@@ -126,7 +126,6 @@ public class JobMetrics_MiscTest extends TestInClusterSupport {
             fail("job didn't fail");
         } catch (Exception e) {
             assertContains(e.toString(), expected.toString());
-            e.printStackTrace();
         }
 
         assertEmptyJobMetrics(job, true);

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/JobMetrics_MiscTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/metrics/JobMetrics_MiscTest.java
@@ -46,7 +46,6 @@ import java.util.function.Function;
 import static com.hazelcast.jet.core.Edge.between;
 import static com.hazelcast.jet.core.JobStatus.RUNNING;
 import static com.hazelcast.jet.core.JobStatus.SUSPENDED;
-import static com.hazelcast.jet.core.TestUtil.assertExceptionInCauses;
 import static com.hazelcast.jet.core.metrics.JobMetrics_BatchTest.JOB_CONFIG_WITH_METRICS;
 import static com.hazelcast.jet.impl.util.ExceptionUtil.peel;
 import static java.util.Collections.singletonList;
@@ -116,17 +115,18 @@ public class JobMetrics_MiscTest extends TestInClusterSupport {
     @Test
     public void when_jobFailedBeforeStarted_then_emptyMetrics() {
         DAG dag = new DAG();
-        RuntimeException exc = new RuntimeException("foo");
+        RuntimeException expected = new RuntimeException("foo");
         // Job will fail in ProcessorSupplier.init method, which is called before InitExecutionOp is
         // sent. That is before any member ever knew of the job.
-        dag.newVertex("v1", new MockPS(MockP::new, 1).setInitError(exc));
+        dag.newVertex("v1", new MockPS(MockP::new, 1).setInitError(expected));
 
         Job job = jet().newJob(dag, JOB_CONFIG_WITH_METRICS);
         try {
             job.join();
             fail("job didn't fail");
         } catch (Exception e) {
-            assertExceptionInCauses(exc, e);
+            assertContains(e.toString(), expected.toString());
+            e.printStackTrace();
         }
 
         assertEmptyJobMetrics(job, true);

